### PR TITLE
Add JWT

### PR DESCRIPTION
**Pull request by EdvardSolomon**

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  secretOrKey: "SECRETVERYHARD",
+  algoritms: ["RS256"],
+};
+
+const strategy = new JwtStrategy(options, (payload, done) => {
+  const repository = getConnection().getRepository("person");
+  const user = repository.findOneBy({ id: payload.id });
+  if (user) {
+    return done(null, user);
+  } else {
```
Comment (antowkabond) [inline config/passport.js:16]: No need else

File: config/passport.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const options = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: "SECRETVERYHARD",
+  algoritms: ["RS256"],
+};
+
+const strategy = new JwtStrategy(options, (payload, done) => {
+  const repository = getConnection().getRepository("person");
+  const user = repository.findOneBy({ id: payload.id });
```
Comment (antowkabond) [inline config/passport.js:13]: Move this logic to service layer

File: src/services/userService.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+const bcrypt = require("bcrypt");
 
 class UserService {
   async create(user) {
+    const repository = getConnection().getRepository("person");
     user.id = uuid.v4();
-    var postRepository = await getConnection().getRepository("person");
-    postRepository.save(user);
+    user.password = await bcrypt.hash(user.password, 3);
+    user.roleid = 1;
```
Comment (antowkabond) [inline src/services/userService.js:11]: create constant cals `DEFAULT_ROLE_ID/ USER_ROLE=1 ` or make it not required field

File: src/services/userService.js [hunk 4]
```diff
@@ -1,11 +1,15 @@
 const uuid = require("uuid");
 const { getConnection } = require("typeorm");
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
 
 class UserService {
   async create(user) {
+    const repository = getConnection().getRepository("person");
     user.id = uuid.v4();
```
Comment (antowkabond) [inline src/services/userService.js:9]: Add this to models hooks 

https://github.com/typeorm/typeorm/blob/master/docs/listeners-and-subscribers.md#beforeinsert

File: src/services/userService.js [hunk 5]
```diff
@@ -32,6 +36,24 @@ class UserService {
     userToDelete.isdeleted = 1;
     repository.save(userToDelete);
   }
+  async login(login, password) {
+    const repository = getConnection().getRepository("person");
+    const user = await repository.findOneBy({ login: login });
+    if (!user) {
+      throw ApiError.BadRequest("Пользователь не был найден!");
```
Comment (antowkabond) [inline src/services/userService.js:43]: no svino languange at all!

File: src/controllers/users.js [hunk 6]
```diff
@@ -26,5 +26,10 @@ class UserControlers {
     await userService.update(id, updateInfo);
     res.send(`User with the id ${id} has been updated`);
   }
+  async loginUser(req, res) {
+    const { login, password } = req.body;
+    const userToken = await userService.login(login, password);
```
Comment (antowkabond) [inline src/controllers/users.js:31]: If `userService.login` throw error how do you handle it?

**Review by anton-makarov-photobox — APPROVED** [review]

**Pull request merged**